.github/workflows/ember.yml

```diff
@@ -0,0 +1,370 @@
+            staging:${{secrets.STAGING_STRIPE_PUBLISHABLE_ID}}
+            production:${{secrets.PRODUCTION_STRIPE_PUBLISHABLE_ID}}
+          GOOGLE_API_KEY: |
+            testing:${{secrets.DEVELOPMENT_GOOGLE_API_KEY}}
+            development:${{secrets.DEVELOPMENT_GOOGLE_API_KEY}}
+            staging:${{secrets.STAGING_GOOGLE_API_KEY}}
+            production:${{secrets.PRODUCTION_GOOGLE_API_KEY}}
+          GOOGLE_CLIENT_ID: |
+            testing:${{secrets.DEVELOPMENT_GOOGLE_CLIENT_ID}}
+            development:${{secrets.DEVELOPMENT_GOOGLE_CLIENT_ID}}
+            staging:${{secrets.STAGING_GOOGLE_CLIENT_ID}}
+            production:${{secrets.PRODUCTION_GOOGLE_CLIENT_ID}}
+
+      # TODO: Get working tests one day
+      # - run: npm test
+
+      - name: "Ember CLI Install"
+        run: |
+          npm install -g ember-cli@3.16
+          which ember || echo "Could not find 'ember'"
+          which ember
+
+      - name: "Ember Build"
+        env:
+          NEWRELIC_LICENSE_KEY: "DEMO"
+          BUILD: ${{ github.run_number }}
+        run: |
+          cd linced-*
+
+          # Verify the env vars required are present
+          if [ -z "$STRIPE_CLIENT_ID" ]; then echo "STRIPE_CLIENT_ID unset"; else echo "STRIPE_CLIENT_ID set: $(echo $STRIPE_CLIENT_ID | wc --chars) characters"; fi
+          if [ -z "$STRIPE_PUBLISHABLE_ID" ]; then echo "STRIPE_PUBLISHABLE_ID unset"; else echo "STRIPE_PUBLISHABLE_ID set: $(echo $STRIPE_PUBLISHABLE_ID | wc --chars) characters"; fi
+          if [ -z "$GOOGLE_API_KEY" ]; then echo "GOOGLE_API_KEY unset"; else echo "GOOGLE_API_KEY set: $(echo $GOOGLE_API_KEY | wc --chars) characters"; fi
+          if [ -z "$GOOGLE_CLIENT_ID" ]; then echo "GOOGLE_CLIENT_ID unset"; else echo "GOOGLE_CLIENT_ID set: $(echo $GOOGLE_CLIENT_ID | wc --chars) characters"; fi
+
+          # Convert the name to UPPER_SNAKE_CASE
+          export PREFIX=$(echo "${{ github.event.repository.name }}" | tr '-' '_' | tr '[:lower:]' '[:upper:]' )
+          echo "Prefix exported as: '${PREFIX}'"
+
+          # Prefix standardised ENV vars provided above
+          export ${PREFIX}_STRIPE_CLIENT_ID=${STRIPE_CLIENT_ID}
+          export ${PREFIX}_STRIPE_PUBLISHABLE_ID=${STRIPE_PUBLISHABLE_ID}
+          export ${PREFIX}_GOOGLE_API_KEY=${GOOGLE_API_KEY}
+          export ${PREFIX}_GOOGLE_CLIENT_ID=${GOOGLE_CLIENT_ID}
+
+          # Create build version
+          export BUILD_NUM=$(( $BUILD + 10000 ))
+          export LINCED_BUILD_VERSION=$( echo "${{ github.ref_name }}.${BUILD_NUM}" | tr '/' '-')
+          echo "Linced Build Version: ${LINCED_BUILD_VERSION}"
+
+          # Install
+          ember build --environment ${{ github.ref_name }}
+
+          # Post install stuff and things.
+          cd dist/assets
+          mkdir update
+          cp linced-*.* update/ || true
+          cp vendor-*.* update/ || true
+
+
+      - name: "Create Sentry Release"
+        run: echo "Not Implemented"
```

Comment:
:feelsbadman:

Reply:
This is such a bitch of a thing....Look at the monstrosity required to match the *.js and *.map names because ember fudges it all up....
https://github.com/linc-technologies/pipelines/blob/main/ember.jsonnet#L323

Would kill for it to generate nicely, I do wonder how easy it will be to add that into a run step....